### PR TITLE
Use form hook

### PR DESCRIPTION
**Pull request by AlcaldeGlobant**
This is an idea to improve performance by changing the input structure I personally like this

### CODE REVIEW

File: src/hooks/useForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    }
+    return errors;
+  };
+
+  const getValidatorsByKey = (key: string) => {
+    return fieldsSettings[key].validators || [];
+  };
+
+  const [values, setValues] = useState<Dictionary<string>>(
+    convertArrayToObject(fieldsSettings, (formInput) => formInput.value || '')
```
Comment (Stefafrontani) [inline src/hooks/useForm/index.tsx:58]: So, we have a function that is called convertArrayToObject. From this name, i assumed that we should pass an array as an argument, but checking the types of fieldSettings i see that it is already an object `fieldSettings: Dictionary<FormInput>`.
Digging a little bit more, it turns out that you need an object to be passed in to this function in order to make Object.entries work as i think it is expected to work. (notice that `Object.entries['stefano', 'carlos']` will not throw an error but [ ["0", "stefano"], ["1", "carlos"]  ] which i think its not what we want here).
I think we should review those data structures (or re-naming some variables maybe) to make some things a little less confusing

File: src/hooks/useForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+// Interfaces END
+
+// Hook START
+const useForm = (fieldsSettings: Dictionary<FormInput>) => {
+  const convertArrayToObject = (
+    dataSource: Dictionary<FormInput>,
+    createValue: ParameterValueFunction<any>
+  ) =>
+    Object.entries(dataSource).reduce(
```
Comment (Stefafrontani) [inline src/hooks/useForm/index.tsx:30]: I have the feeling that we are using reduce just for its own iteration functionality. Here you start with an object of n properties (n = number of [properties/values] ). You convert it into an array of the same length n (Object.entries(...)). Finally you map that array and construct an object from its [key,value] pairs.
I don't see why dont we just use a loop instead:

```
let obj = {};
for (let [key, value] of Object.entries(fieldsSettings)) {
   object[key] = createValue(value, key)
}
return object;
```
 A little comment: before fixing this comment please read the comment related to some data structures.

File: src/containers/Contact/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        {/* Name field*/}
+        <label>Name: </label>
+        <br />
+        <input
+          type="text"
+          name="name"
+          onChange={onChange}
+          value={values.name}
+        />
+        <ul>
```
[the remark this answers is not in the document]
Comment (AlcaldeGlobant) [reply src/containers/Contact/index.tsx:40]: It is not necessary, by default it is an empty array

File: src/containers/Contact/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        <label>Email: </label>
+        <br />
+        <input
+          type="email"
+          name="email"
+          onChange={onChange}
+          value={values.email}
+        />
+        <ul>
+          {errors.email.map((error) => (
```
[the remark this answers is not in the document]
Comment (AlcaldeGlobant) [reply src/containers/Contact/index.tsx:56]: It is not necessary, by default it is an empty array

File: src/containers/Contact/index.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            <li key={error}>Email {error}</li>
+          ))}
+        </ul>
+        <br />
+        {/* Comment field*/}
+        <label>Comment: </label>
+        <br />
+        <textarea name="comment" onChange={onChange} value={values.comment} />
+        <ul>
+          {errors.comment.map((error) => (
```
[the remark this answers is not in the document]
Comment (AlcaldeGlobant) [reply src/containers/Contact/index.tsx:66]: It is not necessary, by default it is an empty array

File: src/hooks/useForm/index.tsx (hunk 2 above)
Comment (AlcaldeGlobant) [reply src/hooks/useForm/index.tsx:30]: I loved this comment, you are right, when the input data structure changes I forgot to look at this method in depth so I agree with you to make this change.

File: src/hooks/useForm/index.tsx (hunk 1 above)
Comment (AlcaldeGlobant) [reply src/hooks/useForm/index.tsx:58]: I want to leave this functionality like this
The intention of doing so is to decouple the functionality and make reuse easier in the future

Comment (Stefafrontani) [reply src/hooks/useForm/index.tsx:56]: Im not talking about the functionality here, my problem is that we are calling convertArrayToObject but NO array is being received in this function.

File: src/containers/Contact/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Stefafrontani) [reply src/containers/Contact/index.tsx:40]: I know, but an empty ul would be rendered anyway.. or am i wrong?

File: src/hooks/useForm/index.tsx (hunk 1 above)
Comment (AlcaldeGlobant) [reply src/hooks/useForm/index.tsx:56]: this will be named as **generateObject**

do you agree? :)

**Pull request closed without merging**